packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -812,6 +859,63 @@ export class Project extends ProjectBase {
       : this._secondaryArtifacts.map((secondaryArtifact) => secondaryArtifact.toArtifactsJSON());
   }
 
+  /**
+   * If configured, set up the VPC-related properties
+   *
+   * Returns the VpcConfig that should be added to the
+   * Lambda creation properties.
+   */
+  private configureVpc(props: ProjectProps): CfnProject.VpcConfigProperty | undefined {
+    if ((props.securityGroups || props.allowAllOutbound !== undefined) && !props.vpc) {
+      throw new Error(`Cannot configure 'securityGroup' or 'allowAllOutbound' without configuring a VPC`);
+    }
+
+    if (!props.vpc) { return undefined; }
+
+    if (props.securityGroups && props.allowAllOutbound !== undefined) {
+      throw new Error(`Configure 'allowAllOutbound' directly on the supplied SecurityGroup.`);
+    }
```

Comment:
Honestly, I'm not sure how I feel about these validations. I understand what you're doing here, but I wonder whether they aren't more irritating than helpful. From what I've seen, we generally don't check for these invariants in props.

However, I won't block the PR on it.

Reply:
Happy to remove, simply copied the approach from another implementation (lambda)